standalone/hsitorySourceTagger.js

```diff
@@ -0,0 +1,48 @@
+// Running this script will cause ZAP to loop through all the entries in the history table.
+// Entries in the table will be tagged (in the "tags" column) in the form:
+// SRC_Proxied, SRC_Manual, SRC_Other
+// The script can be run multiple times, history entries will only be tagged
+// if they don't not already have a tag that starts with TAG_PREFIX as defined below.
+// Author: kingthorin+owaspzap@gmail.com
+// 20160207: Initial release
+
+extHist = org.parosproxy.paros.control.Control.getSingleton().
+      getExtensionLoader().getExtension(org.parosproxy.paros.extension.history.ExtensionHistory.NAME);
+
+TAG_PREFIX='SRC_';
+
+if (extHist != null) {
+  i=org.zaproxy.zap.extension.script.ScriptVars.getGlobalVar("tagged_ref");// Check for global reference
+  if(i==null) {
+    i=1;// Global reference was null so 1
+  }
+  lastRef=extHist.getLastHistoryId();// Get current max history reference 
+  while (i <= lastRef) {// Loop through the history table
+    hr = extHist.getHistoryReference(i);
+      if(i % 10 == 0 | i == 1 | i == lastRef) {// Progress update first, every 10, and last

```

Comment:
The indentation in these statements has too much spaces, compared to above statements.